pkg/operator/secretannotator/vsphere/reconciler.go

```diff
@@ -129,6 +144,17 @@ func (r *ReconcileCloudCredSecret) Reconcile(request reconcile.Request) (reconci
 }
 
 func (r *ReconcileCloudCredSecret) validateCloudCredsSecret(secret *corev1.Secret) error {
+	// We only support passthrough so make sure the operator mode is either unset or "passthrough"
+	mode, _, _, err := utils.GetOperatorConfiguration(r.Client, r.Logger)
+	if err != nil {
+		return err
+	}
+	if mode != "" && mode != operatorv1.CloudCredentialsModePassthrough {
```

Comment:
Not something that we should tackle for this PR, but this is also something that should cause the operator to be degraded.